### PR TITLE
reference/performance: update and add optimizer hints

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? 

1. update some optimizer hints
2. add new optimizer hints

### What is the related PR or file link(s)? 
https://github.com/pingcap/docs-cn/pull/1979
https://github.com/pingcap/docs-cn/pull/2031/

### Which version does your change affect? 

3.1 (will update dev later)

### CODE REVIEW

File: v3.1/reference/performance/optimizer-hints.md [hunk 1]
```diff
@@ -12,40 +12,238 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
+
+In the following hint, `MEMORY_QUOTA(1024 MB)` means that the memory usage limit is 1024 MB:
+
+{{< copyable "sql" >}}
+
+```sql
+select /*+ MEMORY_QUOTA(1024 MB) */ * from t;
+```
+
+In addition to this hint, an environment variable called `tidb_mem_quota_query` can also limit the memory usage of a statement.
```
Comment (lilin90) [inline v3.1/reference/performance/optimizer-hints.md]:
```suggestion
In addition to this hint, the `tidb_mem_quota_query` environment variable can also limit the memory usage of a statement.
```

File: v3.1/reference/performance/optimizer-hints.md [hunk 2]
```diff
@@ -12,40 +12,238 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
+
+{{< copyable "sql" >}}
+
+```sql
+select /*+ READ_FROM_STORAGE(TIFLASH[t1], TIKV[t2]) */ t1.a from t t1, t t2 where t1.a = t2.a;
+```
+
+### USE_INDEX_MERGE(t1, idx1, idx2)
+
+The `USE_INDEX_MERGE(t1, idx1, idx2)` hint tells the optimizer to access a specific table with the index merge method. The given list of indexes are optional parameters. If you explicitly specify the list, TiDB selects indexes from the list to merge index scans; if you do not give the list of indexes, TiDB selects indexes from all available indexes to merge index scans.
```
[the remark this answers is not in the document]
Comment (yikeke) [reply v3.1/reference/performance/optimizer-hints.md]: Updated it already.

File: v3.1/reference/performance/optimizer-hints.md [hunk 3]
```diff
@@ -12,40 +12,238 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
+
+Each query and sub-query in a statement corresponds to a different query block, and each query block has its own `QB_NAME`. For example:
+
+{{< copyable "sql" >}}
+
+```sql
+select * from (select * from t) t1, (select * from t) t2;
+```
+
+The above query statement has three query blocks: the outermost `SELECT` corresponds to the first query block, whose `QB_NAME` is `sel_1`; the two sub-queries correspond to the second and the third query block, whose `QB_NAME`s are `sel_2` and `sel_3` respectively. The sequence of the numbers is based on the appearance of `SELECT` from left to right. If `DELETE` and `UPDATE` queries substitute for the two `SELECT` sub-queries, then the `QB_NAME`s are `del_1` and `upd_1`.
```
[the remark this answers is not in the document]
Comment (yikeke) [reply v3.1/reference/performance/optimizer-hints.md]: I rewrite this sentence as "If you replace the two `SELECT` sub-queries with a `DELETE` query and a `UPDATE` query, then their `QB_NAME`s will be `del_1` and `upd_1`". Is that clearer?

[the remark this answers is not in the document]
Comment (yikeke) [reply v3.1/reference/performance/optimizer-hints.md]: Updated it according to the latest changes in the original Chinese. PTAL again.

**Comment by yikeke** [comment]
All comments are addressed. PTAL again. @lilin90 @eurekaka

File: v3.1/reference/performance/optimizer-hints.md [hunk 4]
```diff
@@ -6,46 +6,252 @@ category: reference
...
+select /*+ QB_NAME(QB1) */ * from (select * from t) t1, (select * from t) t2;
+```
+
+This hint means that the `SELECT` query block's name is specified to `QB1`, which makes `QB1` and the default name `sel_1` both valid for the query block.
+
+> **Note:**
+>
+> In the above example, if the hint specifies the `QB_NAME` to `sel_2` and does not specify a new `QB_NAME` for the original second `SELECT` query block, then `sel_2` becomes an invalid name for the second `SELECT` query block.
+
+### `@QB_NAME`
```
Comment (lilin90) [inline v3.1/reference/performance/optimizer-hints.md]:
```suggestion
### @QB_NAME
```

File: dev/reference/performance/optimizer-hints.md [hunk 5]
```diff
@@ -6,48 +6,252 @@ category: reference
...
+select /*+ QB_NAME(QB1) */ * from (select * from t) t1, (select * from t) t2;
+```
+
+This hint means that the `SELECT` query block's name is specified to `QB1`, which makes `QB1` and the default name `sel_1` both valid for the query block.
+
+> **Note:**
+>
+> In the above example, if the hint specifies the `QB_NAME` to `sel_2` and does not specify a new `QB_NAME` for the original second `SELECT` query block, then `sel_2` becomes an invalid name for the second `SELECT` query block.
+
+### `@QB_NAME`
```
Comment (lilin90) [inline dev/reference/performance/optimizer-hints.md]: The inline code format isn't used for all other headings of the same level. Suggest removing it to keep the format consistent.
```suggestion
### @QB_NAME
```

Comment (yikeke) [reply dev/reference/performance/optimizer-hints.md]: Sounds good.

**Pull request merged**